ci/kind/test-e2e-kind.sh

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 Antrea Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script is required for Antrea to work properly in a Kind cluster on Linux. It takes care of
+# disabling TX hardware checksum offload for the veth interface (in the host's network namespace) of
+# each Kind Node. This is required when using OVS in userspace mode. Refer to
+# https://github.com/vmware-tanzu/antrea/issues/14 for more information.
+
+# The script uses the antrea/ethtool Docker image (so that ethtool does not need to be installed on
+# the Linux host).
+
+set -eo pipefail
+
+TESTBED_CMD=$(dirname $0)"/kind-setup.sh"
+YML_CMD=$(dirname $0)"/../../hack/generate-manifest.sh"
+COMMON_IMAGES="busybox nginx antrea/antrea-ubuntu:latest"
+
+function quit {
+  if [[ $? != 0 ]]; then
+    echo " Test failed cleaning testbed"
+    $TESTBED_CMD destroy kind
+  fi
+}
+trap "quit" INT EXIT
+
+function run_test {
+  mode=$1
+  args=$2
+  echo "create test bed with args $args"
+  eval "timeout 600 $TESTBED_CMD create kind --antrea-cni false $args"
+  $YML_CMD --kind  --encap-mode $mode | docker exec -i kind-control-plane dd of=/root/antrea.yml
+  sleep 1
+  go test -v -timeout=20m github.com/vmware-tanzu/antrea/test/e2e -provider=kind
+  $TESTBED_CMD destroy kind
+}
+
+docker pull busybox
+docker pull nginx
+
+echo "======== Test encap mode =========="
```

Comment:
Hi antonin, 

I has some reservations about sharing network resources, container resource to run all test cases at the same time. Tests may be interfering with each other or vm may running out of resources with three k8s clusters running.

I have a subsequent change ready for PR that uses a single test bed for all three mode, and each additional network mode only run a small set of network tests. This would also address the time run issue. Hope this will be OK with u.

resolve this for now, please feel free to re-open if this does not adequately address ur concern.

Reply:
I'm not asking for tests to be run on the same cluster. It is a very easy change to create 3 Github jobs that can run in parallel instead of running the tests sequentially in the same Github job. Similarly to how we run e2e tests in parallel with unit tests. Github does not impose a limit on the number of parallel jobs (unlike some other CI systems), it will let you use whichever resources are available.

Reply:
got it.
break kind.yml into three cases one for each encap mode.

Reply:
Antonin, 

what is expected test case? 

For instance, if I rename kind test to

"-name E2e encap tests on a Kind cluster on Linux"

the test would run, but there would be outstanding expectation, some thinkg

expect E2e tests on a Kind cluster on Linux ...., and as result tests requirements are not fulfilled.

change the kind test back to
"-name E2e tests on a Kind cluster on Linux" solve the problem.

I did not find anywhere in repro making reference to expected tests.

Thx, Su

Reply:
I believe you should keep everything inside kind.yml. The name of the "workflow"  is still "Kind", but you can have 3 entries in the "jobs" array:
 * E2e tests on a Kind cluster on Linux (encap)
 * E2e tests on a Kind cluster on Linux (hybrid)
 * E2e tests on a Kind cluster on Linux (noEncap)

These jobs should execute in parallel if Github as enough compute resources available.
But maybe I'm missing the question?